pallets/connectors/src/inbound.rs

```diff
@@ -0,0 +1,263 @@
+	/// `DomainLocator` account of the origination domain beforehand.
+	pub fn handle_tranche_tokens_transfer(
+		pool_id: PoolIdOf<T>,
+		tranche_id: TrancheIdOf<T>,
+		sending_domain: DomainAddress,
+		receiver: T::AccountId,
+		amount: <T as Config>::Balance,
+	) -> DispatchResult {
+		ensure!(!amount.is_zero(), Error::<T>::InvalidTransferAmount);
+
+		ensure!(
+			T::Permission::has(
+				PermissionScope::Pool(pool_id),
+				receiver.clone(),
+				Role::PoolRole(PoolRole::TrancheInvestor(tranche_id, Self::now())),
+			),
+			Error::<T>::UnauthorizedTransfer
+		);
+
+		let invest_id = Self::derive_invest_id(pool_id, tranche_id)?;
+
+		T::Tokens::transfer(
+			invest_id.into(),
+			&Domain::convert(sending_domain.domain()),
+			&receiver,
+			amount,
+			false,
+		)?;
+
+		Ok(())
+	}
+
+	/// Increases an existing investment order of the investor.
+	///
+	/// Directly mints the additional investment amount into the investor
+	/// account.
+	pub fn increase_invest_order(
+		pool_id: PoolIdOf<T>,
+		tranche_id: TrancheIdOf<T>,
+		investor: T::AccountId,
+		currency_index: GeneralCurrencyIndexOf<T>,
+		amount: <T as Config>::Balance,
+	) -> DispatchResult {
+		// Retrieve investment details
+		let invest_id: T::TrancheCurrency = Self::derive_invest_id(pool_id, tranche_id)?;
+		let currency = Self::try_get_payment_currency(invest_id.clone(), currency_index)?;
+
+		// Determine post adjustment amount
+		let pre_amount = T::ForeignInvestment::investment(&investor, invest_id.clone())?;
+		let post_amount = pre_amount.ensure_add(amount)?;
+
+		// Mint additional amount
+		T::Tokens::mint_into(currency, &investor, amount)?;
+
+		T::ForeignInvestment::update_investment(&investor, invest_id, post_amount)?;
+
+		Ok(())
+	}
+
+	/// Decreases an existing investment order of the investor.
+	///
+	/// Directly burns the decreased investment amount from the investor
+	/// account.
+	///
+	/// Initiates a return `ExecutedDecreaseInvestOrder` message to refund the
+	/// decreased amount on the source domain. The dispatch of this message is
+	/// delayed until the execution of the investment, e.g. at least until the
+	/// next epoch transition.
+	pub fn do_decrease_invest_order(
+		pool_id: PoolIdOf<T>,
+		tranche_id: TrancheIdOf<T>,
+		investor: T::AccountId,
+		currency_index: GeneralCurrencyIndexOf<T>,
+		amount: <T as Config>::Balance,
+	) -> DispatchResult {
+		// Retrieve investment details
+		let invest_id: T::TrancheCurrency = Self::derive_invest_id(pool_id, tranche_id)?;
+		let currency = Self::try_get_payment_currency(invest_id.clone(), currency_index)?;
+
+		// Determine post adjustment amount
+		let pre_amount = T::ForeignInvestment::investment(&investor, invest_id.clone())?;
+		let post_amount = pre_amount.ensure_sub(amount)?;
+
+		T::ForeignInvestment::update_investment(&investor, invest_id, post_amount)?;
+
+		// Burn decreased amount
+		T::Tokens::burn_from(currency, &investor, amount)?;
```

Comment:
Here we keep the burn, below we removed the transfer.

Reply:
Sorry, this slipped through. Will handle this.